admin/src/js/modules/ajax-download.js

```diff
@@ -2,18 +2,37 @@
 
   'use strict';
 
-  const DEFAULT_FILE_NAME = 'download';
+  const getFileName = (headers) => {
+    const DEFAULT_FILE_NAME = 'download';
+
+    const contentDisposition = headers.get('content-disposition');
+    if (!contentDisposition) {
+      return DEFAULT_FILE_NAME;
+    }
+
+    const fileName = contentDisposition.split('filename=')[1];
+    if (!fileName) {
+      return DEFAULT_FILE_NAME;
+    }
+
+    return fileName;
+  };
 
   /**
    * Prompts the user to download a file given a url.
    */
   exports.download = function(url) {
-    const element = document.createElement('a');
-    element.setAttribute('href', url);
-    element.setAttribute('download', DEFAULT_FILE_NAME);
-    element.style.display = 'none';
-    document.body.appendChild(element);
-    element.click();
-    document.body.removeChild(element);
+    return fetch(url)
```

Comment:
Not a huge deal, but we did change up the logic quite a bit in this file.  Would be nice to have some unit tests if they are feasible.

Reply:
That being said, why did we need to change up the default file name here?  (obviously the current approach is not ideal, but is this just an enhancement on top of our User Devices functionality, or did we need this for some reason?)

Reply:
This is more of an enhancement. With this approach, we know when the download starts and when it ends. It enables us to give feedback to the user and prevent them from requesting a duplicate export while one is being processed.

Reply:
Sorry forgot to answer your second point about the file name. The default file name is still the same but we need to explicitly set the file name with this API because it doesn't automatically extract it from the `Content-Disposition` header, so we do it ourselves. The behavior is the same: if there is such a header, it uses the file name provided, otherwise it falls back to the default file name.

Reply:
> to support the $scope.exporting logic to disable the download buttons wile the current request is processing

Ah okay, so yes, using `fetch` and return a promise makes sense here.